timepiece/contracts/models.py

```diff
@@ -60,6 +60,30 @@ def get_admin_url(self):
     def get_absolute_url(self):
         return reverse('view_contract', args=[self.pk])
 
+    def get_noncontract_entries(self, entries=None):
+        """
+        Given a set of entries, exclude those included in any contract affiliated
+        with any project associated with this contract.
+        """
+        contracts = []
+        for project in self.projects.all():
+            contracts.extend(project.contracts.all())
+
+        for contract in contracts:
+            entries.exclude(
+                project__in=contract.projects.all(),
+                start_time__gte=contract.start_date,
+                end_time__lt=contract.end_date + relativedelta(days=1)
+                )
+        return entries
+
+    @property
+    def pre_launch_entries(self):
+        pce = Entry.objects.filter(
```

Comment:
very ticky tacky, but can we rename `pce` to `entries` where it is used throughout this PR.  Or `projectcontract_entries`, if you want to be verbose.